### PR TITLE
Changes the relationships format to a map inside endpoint element.

### PR DESCRIPTION
**Pull request by rosogon**
A relationship is now like:

    requirements:
    - endpoint:
        node: targetNode
        type: relationType
        properties:
          prop.name: connurl

when previously was defined like:

    requirements:
    - endpoint: targetNode
      type: relationType
      properties:
        prop.name: connurl

This change facilitates the generic handling of relationships in the DAM
generation.

### CODE REVIEW

**Comment by rosogon** [comment]
@kiuby88 : Could you have a look?

File: planner/aamwriter/src/main/resources/examples/atos_case_study_aam.yml [hunk 1]
```diff
@@ -1,110 +1,112 @@
...
+topology_template:
+  node_templates:
+    www:
+      type: sc_req.www
+      properties:
+        language: JAVA
+      requirements:
+      - endpoint:
+          node: webservices
+          type: seaclouds.relation.connection.endpoint.host
```
Comment (kiuby88) [inline planner/aamwriter/src/main/resources/examples/atos_case_study_aam.yml]: could you add `prop.name`?

File: planner/aamwriter/src/main/resources/examples/atos_case_study_aam.yml [hunk 2]
```diff
@@ -1,110 +1,112 @@
...
+        language: JAVA
+        location: ''
+        java_version:
+          constraints:
+          - greater_or_equal: '7'
+          - less_or_equal: '8'
+      requirements:
+      - endpoint:
+          node: db1
+          type: seaclouds.relations.databaseconnections.jdbc
```
Comment (kiuby88) [inline planner/aamwriter/src/main/resources/examples/atos_case_study_aam.yml]: could you add `prop.name`?

File: planner/aamwriter/src/main/resources/examples/nuro_case_study_aam.yml [hunk 3]
```diff
@@ -1,65 +1,66 @@
...
+        language: PHP
+        autoscale: false
+        php_version:
+          constraints:
+          - greater_or_equal: '5.1'
+          - less_or_equal: '5.5'
+      requirements:
+      - endpoint:
+          node: db1
+          type: seaclouds.relations.databaseconnections.jdbc
```
Comment (kiuby88) [inline planner/aamwriter/src/main/resources/examples/nuro_case_study_aam.yml]: could you add `prop.name`?

File: planner/aamwriter/src/main/resources/examples/web-chat_aam.yml [hunk 4]
```diff
@@ -1,79 +1,80 @@
...
+      type: sc_req.Chat
+      properties:
+        language: JAVA
+      artifacts:
+      - wars.root: http://www.seaclouds.eu/artifacts/chat-webApplication.war
+        type: tosca.artifacts.File
+      requirements:
+      - endpoint:
+          node: MessageDatabase
+          type: seaclouds.relations.databaseconnections.jdbc
```
Comment (kiuby88) [inline planner/aamwriter/src/main/resources/examples/web-chat_aam.yml]: could you add `prop.name`?

**Comment by kiuby88** [comment]
Good work @rosogon, it looks good to me. I have just added a pair of minor comments

**Comment by rosogon** [comment]
Ok. Comments addressed. I'll squash and merge.

**Pull request merged**